### ISSUE TITLE
blacklist fails to match against a pattern I added

### ISSUE THREAD

**Issue by StentMan** [Robbendebiene/Gesturefy#542]
### Describe your matter in detail
This site uses it's own set or right-click gestures which conflict with Gesturefy. Gesturefy overrides the site's features. Yes, I use right-click as the activation button.
How to reproduce:
1. Add the following blacklist pattern: *://www.chess.com/game/live/*
2. Register a free account at www.chess.com and start playing a 5 minute game. The URL on the actual game page should look something like this: https://www.chess.com/game/live/5477498819. If it doesn't, then you're redirected to the old game page who's blacklist pattern (*://www.chess.com/live?*) works just fine.
3. Try doing gestures - for me, at least, the gestures work so the blacklist isn't working as expected. Maybe this only happens on 2nd or 3rd game? not sure. When the game is done  just click the new 5 min game button on the right.

Fun fact: if I refresh the game page gestures no longer works.

### Your System
- Operating system: Windows 10
- Firefox version: 80.0.1
- Gesturefy version: 3.0.7

**Comment by Robbendebiene** [comment]
You should use this blacklist entry: `*://www.chess.com/game/live/*`

**Comment by StentMan** [comment]
Yes, that's the entry I used. Sorry for the mixup, I don't know why the asterix characters were removed from my post. probably should've escaped them, somehow.

**Comment by Robbendebiene** [comment]
Yeah I see. anything surrounded by asterix is transformed into italic style by githubs markdown.

Okay I haven't created an account, but if I just visit https://www.chess.com/game/live/5477498819 with the blacklist entry applied gestures are disabled correctly for me. 

> As I wrote, this site is somewhat unusual, as it uses its own set of right-click gestures: a simple right-click to cancel pre-moves, right-click drag to draw arrows on the chess board.

I'm also able to do this.



> Fun fact: if I refresh the game page gestures no longer works.

This lets me suspect that when you start playing a new game the website does not change entirely (it basically only changes and loads some new content, but not an entire new page). After that it probably adjusts the URL. In this case Gesturefy would miss that you are on a new URL. Can you confirm this? If this would be the problem I should be able to fix this.

**Comment by StentMan** [comment]
> This lets me suspect that when you start playing a new game the website does not change entirely (it basically only changes and loads some new content, but not an entire new page). After that it probably adjusts the URL. In this case Gesturefy would miss that you are on a new URL. Can you confirm this? If this would be the problem I should be able to fix this.

Confirm this how? I'm not sure what kind of actions/observations you'd like me to perform in order to confirm that?
And thank you for looking into this.

**Comment by Robbendebiene** [comment]
Can you send me a short screen recording of the steps you've described above, so I can see when the URL changes in the address bar? You basically can also observe this your own if you look carefully at the browsers reloading symbol. If it does **not** change shortly to an X symbol after you've started a new game (in other words the browser changed the URL to https://www.chess.com/game/live/XXX) then my assumption is correct.

**Comment by StentMan** [comment]
Sure, I can do that.

**Comment by Robbendebiene** [comment]
Thanks for the video! Exactly as I've supposed.
I should be able to fix this. In the mean time you can only use a disable button (alt, ctrl etc.) as a workaround.


> Edit2: haha, no PM-ing. it's not a "proper" forum. Here's the link: https://app.box.com/s/m0wmyqqwjcej0pxiu1sdqhibkib44fys

Yep, sadly this is not possible on Github. (You can now remove the video or link if you want to)

> Not that it matters, but I won that game

Not that it matters, but only 5 minutes for a chess round would be way too hectic for me :D

**Comment by StentMan** [comment]
I won't prioritize this bug if I were you. this is a new game page chess.com is rolling out or, maybe, it's a work in progress. I just played today and the game redirected me to the old game page who's url looks something like this: `https://www.chess.com/live?#g=5486375113` and for that page the blacklist rule `*://www.chess.com/live?*` works just fine.
This has happened in the past, they went back and forth with this new page and back 2-3 times already.

**Comment by StentMan** [comment]
Did you fix it? or, maybe, they made some changes to the page? Either way, after a long while I played some chess, again on chess.com and the blacklist pattern worked.
You can mark this as fixed (or not a bug?).

Thanks for looking into this.

**Comment by Robbendebiene** [comment]
Thanks for the update. I implemented the fix, but it's not shipped yet. They probably changed something on their website.
I'm reopening this ticket as a reminder and I'll close it again when the fix is shipped.
